### PR TITLE
Makes species with the NO_BLOOD flag immune to most chemical stuns and effects

### PR DESCRIPTION
**Pull request by Alberyk**
Makes species with NO_BLOOD (skeletons, shadows, golems, etc) immune to most knock down/stuns/etc chems, no more injecting skeletons with chloral hydrate.

### CODE REVIEW

**Comment by Lohikar** [comment]
It would be better to hook into what calls affect_blood instead of adding a check into literally every affect_blood override.

**Comment by Alberyk** [comment]
No, because, I still want certain, possible, chems to have effects in certain species.

**Comment by Lohikar** [comment]
How is `affect_blood` ever valid in a species that does not have blood?

**Comment by Alberyk** [comment]
Dionae have no blood, still chems such as plant b gone can affect them.

**Comment by Lohikar** [comment]
Ah, that makes sense.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm [hunk 1]
```diff
@@ -116,13 +119,14 @@
...
 	..()
-	if(ishuman(M))
-		var/mob/living/carbon/human/H = M
-		if(H.stat != 1)
-			if(H.losebreath >= 10)
-				H.losebreath = max(10, H.losebreath - 10)
-			H.adjustOxyLoss(2)
-			H.Weaken(10)
+	var/mob/living/carbon/human/H = M
+	if(istype(H) && (H.species.flags & NO_BLOOD))
```
Comment (skull132) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm]: This is not the logical equivalent of what was before. The lower if will still active if mob is **not** human, which is bad.

Use this one instead: `if (!istype(H) || (H.species.flags & NO_BLOOD))`

It'll stop if the mob is not human OR if the mob is human but has NO_BLOOD.

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm [hunk 2]
```diff
@@ -135,13 +139,14 @@
...
 	..()
-	if(ishuman(M))
-		var/mob/living/carbon/human/H = M
-		if(H.stat != 1)
-			if(H.losebreath >= 10)
-				H.losebreath = max(10, M.losebreath-10)
-			H.adjustOxyLoss(2)
-			H.Weaken(10)
+	var/mob/living/carbon/human/H = M
+	if(istype(H) && (H.species.flags & NO_BLOOD))
```
Comment (skull132) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Toxins.dm]: Same as above.

**Comment by MarinaGryphon** [comment]
Does this stop sleepers from affecting them? For some strange reason, IPCs can still get incapacitated by sleepers, though that may be unrelated to the `NO_BLOOD` thing.

**Comment by Alberyk** [comment]
Fixed them both, also, only mecha sleepers put mobs to sleep, which has nothing to do with this.

**Pull request merged**